src/modules/Dropdown/Dropdown.js

```diff
@@ -729,8 +729,7 @@ export default class Dropdown extends Component {
     this.setValue(newValue)
     this.setSelectedIndex(value)
 
-    const optionSize = _.size(this.getMenuOptions())
-    if (!multiple || isAdditionItem || optionSize === 1) this.clearSearchQuery()
+    this.clearSearchQuery()
```

Comment:
This change is absolutely valid, we did this in #2162 and even had an comment about this https://github.com/Semantic-Org/Semantic-UI-React/pull/2162#issuecomment-355130565.

See our decision https://github.com/Semantic-Org/Semantic-UI-React/pull/2162#issuecomment-334096165. I checked examples manually, LGTM.